backend/ingestion/extract_scheduled_gtfs.py

```diff
@@ -0,0 +1,304 @@
+CTA_URL = "https://www.transitchicago.com/downloads/sch_data/google_transit.zip"
+METRA_URL = "https://schedules.metrarail.com/gtfs/schedule.zip"
+
+## NEW YORK
+# Source: https://new.mta.info/developers
+MTA_SUBWAY_URL = "http://web.mta.info/developers/data/nyct/subway/google_transit.zip"
+# Bus data is divided up by borough, with a sixth "MTA Bus Company" feed for
+# select lines in Brooklyn/Queens. "shapes.txt" file for each of these is huge
+BRONX_BUS_URL = "http://web.mta.info/developers/data/nyct/bus/google_transit_bronx.zip"
+BROOKLYN_BUS_URL = (
+    "http://web.mta.info/developers/data/nyct/bus/google_transit_brooklyn.zip"
+)
+MANHATTAN_BUS_URL = (
+    "http://web.mta.info/developers/data/nyct/bus/google_transit_manhattan.zip"
+)
+QUEENS_BUS_URL = (
+    "http://web.mta.info/developers/data/nyct/bus/google_transit_queens.zip"
+)
+STATEN_ISLAND_BUS_URL = (
+    "http://web.mta.info/developers/data/nyct/bus/google_transit_staten_island.zip"
+)
+MTA_BUS_CO_BUS_URL = "http://web.mta.info/developers/data/busco/google_transit.zip"
+
+## PORTLAND
+TRIMET_URL = "http://developer.trimet.org/schedule/gtfs.zip"
+
+
+ALL_PILOT_CITY_URLS = [
+    CTA_URL,
+    METRA_URL,
+    MTA_SUBWAY_URL,
+    BRONX_BUS_URL,
+    BROOKLYN_BUS_URL,
+    MANHATTAN_BUS_URL,
+    QUEENS_BUS_URL,
+    STATEN_ISLAND_BUS_URL,
+    MTA_BUS_CO_BUS_URL,
+    TRIMET_URL,
+]
+
+URL_TO_CITY = {
+    "https://www.transitchicago.com/": {"City": "Chicago", "Agency": "CTA"},
+    "https://schedules.metrarail.com/": {"City": "Chicago", "Agency": "Metra"},
+    "http://web.mta.info/": {"City": "New York", "Agency": "MTA"},
+    "http://developer.trimet.org/": {"City": "Portland", "Agency": "Trimet"},
+}
+
+FILE_TO_PRIMARY_KEY = {
+    # Primary keys as defined in GTFS standard guide
+    # https://gtfs.org/schedule/reference/#stopstxt
+    # TODO: Question -- do we want to create a new primary key that concatenates
+    # city and agency information, or just do JOINs that add city and agency
+    # to the ON condition to make sure all JOINs are within a city-agency system?
+    "routes": ("route_id"),
+    "trips": ("trip_id"),
+    "stops": ("stop_id"),
+    "stop_times": ("trip_id", "stop_sequence"),
+    "shapes": ("shape_id", "shape_pt_sequence"),
+    "transfers": (
+        "from_stop_id",
+        "to_stop_id",
+        "from_trip_id",
+        "to_trip_id",
+        "from_route_id",
+        "to_route_id",
+    ),
+    "shape_geometries": ("route_id"),  # speculative
+}
+
+### Functions
+
+
+def ingest_gtfs_feed(feed_url: str) -> tuple:
+    """
+    Use gtfs_kit library to ingest the most up-to-date scheduled (static) GTFS
+    data from a particular URL, and return each component file as a GeoDataFrame.
+    """
+    url_base = re.findall(r"https?:\/\/[^\/]+\/", feed_url)[0]
+    feed_city = URL_TO_CITY[url_base]["City"]
+    feed_agency = URL_TO_CITY[url_base]["Agency"]
+    feed = gk.read_feed(feed_url, dist_units="km")  # dist_units TBD
```

Comment:
just out of curiosity, is it possible to get miles instead? My sense is that will be a more standard unit for US transit but can't see any problems with it for now

Reply:
yep - can just change `dist_units="km"` to `dist_units="mi"`